batch/sql/jobs-after-update-simple.sql

```diff
@@ -0,0 +1,146 @@
+DELIMITER $$
+
+DROP TRIGGER IF EXISTS jobs_after_update $$
+CREATE TRIGGER jobs_after_update AFTER UPDATE ON jobs
+FOR EACH ROW
+BEGIN
+  DECLARE cur_user VARCHAR(100);
+  DECLARE cur_batch_cancelled BOOLEAN;
+  DECLARE cur_n_tokens INT;
+  DECLARE rand_token INT;
+
+  DECLARE always_run boolean;
+  DECLARE cores_mcpu bigint;
+
+  DECLARE was_marked_cancelled boolean;
+  DECLARE was_cancelled        boolean;
+  DECLARE was_cancellable      boolean;
+
+  DECLARE now_marked_cancelled boolean;
+  DECLARE now_cancelled        boolean;
+  DECLARE now_cancellable      boolean;
+
+  DECLARE was_ready boolean;
+  DECLARE now_ready boolean;
+
+  DECLARE was_running boolean;
+  DECLARE now_running boolean;
+
+  DECLARE was_creating boolean;
+  DECLARE now_creating boolean;
+
+  DECLARE delta_n_ready_cancellable_jobs          int;
+  DECLARE delta_ready_cancellable_cores_mcpu   bigint;
+  DECLARE delta_n_ready_jobs                      int;
+  DECLARE delta_ready_cores_mcpu               bigint;
+  DECLARE delta_n_cancelled_ready_jobs            int;
+
+  DECLARE delta_n_running_cancellable_jobs        int;
+  DECLARE delta_running_cancellable_cores_mcpu bigint;
+  DECLARE delta_n_running_jobs                    int;
+  DECLARE delta_running_cores_mcpu             bigint;
+  DECLARE delta_n_cancelled_running_jobs          int;
+
+  DECLARE delta_n_creating_cancellable_jobs       int;
+  DECLARE delta_n_creating_jobs                   int;
+  DECLARE delta_n_cancelled_creating_jobs         int;
+
+  SELECT user INTO cur_user FROM batches WHERE id = NEW.batch_id;
+
+  SET cur_batch_cancelled = EXISTS (SELECT TRUE
+                                    FROM batches_cancelled
+                                    WHERE id = NEW.batch_id
+                                    LOCK IN SHARE MODE);
+
+  SELECT n_tokens INTO cur_n_tokens FROM globals LOCK IN SHARE MODE;
+  SET rand_token = FLOOR(RAND() * cur_n_tokens);
+
+  SET always_run = old.always_run; # always_run is immutable
+  SET cores_mcpu = old.cores_mcpu; # cores_mcpu is immutable
+
+  SET was_marked_cancelled = old.cancelled OR cur_batch_cancelled;
+  SET was_cancelled        = NOT always_run AND was_marked_cancelled;
+  SET was_cancellable      = NOT always_run AND NOT was_marked_cancelled;
+
+  SET now_marked_cancelled = new.cancelled or cur_batch_cancelled;
+  SET now_cancelled        = NOT always_run AND now_marked_cancelled;
+  SET now_cancellable      = NOT always_run AND NOT now_marked_cancelled;
+
+  # NB: was_cancelled => now_cancelled b/c you cannot be uncancelled
```

Comment:
I'm not sure what this comment was meant to say. Is it that if `was_cancelled = True`, then `now_cancelled` must be True?

Reply:
I replaced `=>` with `implies`